ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,210 @@
-            -   Enter the word 'token' into the 'User Name:' field, then enter your 'Databricks access token' in the 'Password:' field.
-            -   Change the Thrift Transport option to HTTP.
-            -   Click the 'HTTP Options...' button and enter the 'HTTP Path' of your Databricks cluster, then click 'Okay'.
-            -   Click the 'SSL Options...' button and tick the 'Enable SSL' box, then click the 'OK' button.
-            -   ![](../images/odbc-driver-settings.png)
-        c.  Click the 'Test' button to verify the connection has worked. You should see the following message. *If you get an error here, repeat steps 5.e.i -- 5.e.ix again and ensure all the values are correct.*
+A personal access token is is a security measure that acts as an identifier to let Databricks know who is accessing information from the SQL warehouse. Access tokens are usually set for a limited amount of time, so they will need renewing periodically.
 
-        ![](../images/databricks-test-connection.png)
+-   In Databricks, click on your email address in the top right corner, then click 'User settings'
 
-        d.  Click the 'OK' button to exit the 'Test Results' window, then the 'OK' button in the 'Simba Spark ODBC Driver DSN Setup' window.
+-   Go to the 'Developer' tab in the side bar. Next to 'Access tokens', click the 'Manage' button
 
-5.  Connect through RStudio. Watch the below video and view the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect) for methods on connecting to Databricks and querying data from RStudio. 
+::: {align="center"}
+![](../images/databricks-access-tokens.png)
+:::
 
+-   Click the 'Generate new token' button
+
+-   Name the token, then click 'Generate'
+
+::: callout-note
+Note that access tokens will only last as long as the value for the 'Lifetime (days)' field. After this period the token will expire, and you will need to create a new one to re-authenticate.
+:::
+
+-   Make a note of the 'Databricks access token' it has given you
+
+::: callout-warning
+It is very important that you immediately copy the access token that you are given, as you will not be able to see it through Databricks again. If you lose this access token before pasting it into RStudio then you must generate a new access token to replace it.
+:::
+
+------------------------------------------------------------------------
+
+#### Cluster ID
 
 ------------------------------------------------------------------------
 
-# Pulling data into R studio from Databricks
+The Cluster ID is the cluster that you'll be using to run your code.
+
+To obtain this, follow these steps:
+
+-   click Compute in the left and menu, and click on the name of your cluster
+-   under the configuration tab, click on "Advanced options" at the bottom of the page
+-   click on the JDBC/ODBC tab
+-   in HTTP path, copy everything after the last /. This is your Cluster ID 
```

Comment:
As above, it's actually the full path we want and we probably want to rename it "..._PATH" rather than "..._ID".